hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4818,3 +4859,169 @@ func isAPIServerRoute(hcluster *hyperv1.HostedCluster) bool {
 	}
 	return false
 }
+
+func reconcileControlPlanePKIOperatorDeployment(
+	deployment *appsv1.Deployment,
+	openShiftTrustedCABundleConfigMapExists bool,
+	hc *hyperv1.HostedCluster,
+	hcp *hyperv1.HostedControlPlane,
+	cpoPKIImage string,
+	setDefaultSecurityContext bool,
+	sa *corev1.ServiceAccount,
+) error {
+
+	cpoResources := corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("80Mi"),
+			corev1.ResourceCPU:    resource.MustParse("10m"),
+		},
+	}
+	// preserve existing resource requirements for main cpo-pki container
+	mainContainer := hyperutil.FindContainer("control-plane-pki-operator", deployment.Spec.Template.Spec.Containers)
+	if mainContainer != nil {
+		if len(mainContainer.Resources.Requests) > 0 || len(mainContainer.Resources.Limits) > 0 {
+			cpoResources = mainContainer.Resources
+		}
+	}
+
+	deployment.Spec = appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"name": "control-plane-pki-operator",
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"name":                        "control-plane-pki-operator",
+					"app":                         "control-plane-pki-operator",
+					hyperv1.ControlPlaneComponent: "control-plane-pki-operator",
+				},
+			},
+			Spec: corev1.PodSpec{
+				ImagePullSecrets: []corev1.LocalObjectReference{
+					{
+						Name: "pull-secret",
+					},
+				},
+				ServiceAccountName: sa.Name,
+				Containers: []corev1.Container{
+					{
+						Name:            "control-plane-pki-operator",
+						Image:           cpoPKIImage,
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Env: []corev1.EnvVar{
+							{
+								Name: "HOSTED_CONTROL_PLANE_NAMESPACE",
+								ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "metadata.namespace",
+									},
+								},
+							},
+							{
+								Name:  "HOSTED_CONTROL_PLANE_NAME",
+								Value: hcp.Name,
+							},
+						},
+						Command: []string{"/usr/bin/control-plane-pki-operator"},
```

Comment:
traditionally this would be a subcommand of the cpo, see https://github.com/openshift/hypershift/blob/3a42e77041535c8ac8012856d279bc782efcaf3c/control-plane-operator/main.go#L75
and 
https://github.com/openshift/hypershift/blob/3a42e77041535c8ac8012856d279bc782efcaf3c/hypershift-operator/controllers/hostedcluster/ignitionserver/ignitionserver.go#L554-L555

Reply:
@sjenning requested that it not be in this case

Reply:
I assume is to keep library go controller code fully decoupled? sgmt whatever Seths preference.

Reply:
Yeah I think so